### PR TITLE
Feature/base templator and register

### PR DESCRIPTION
**Pull request by Lex-23**

### CODE REVIEW

File: cryptostock_api/cryptostock/notification/models.py [hunk 1]
```diff
@@ -98,3 +99,27 @@ def enable_notification_subscriptions(self):
...
+            return templater_cls
+
+        return inner
+
+    get = classmethod(_templaters.get)
+
+
+class BaseTemplator:
+    def render(self, data: Dict[str, Any]) -> Any:
+        return "Info".format(**data)
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/notification/models.py:120]:
```suggestion
        return f"Event: {data}"
```

File: cryptostock_api/cryptostock/notification/models.py [hunk 2]
```diff
@@ -98,3 +99,27 @@ def enable_notification_subscriptions(self):
...
+
+
+@TemplaterRegister.register(ConsumerType.TELEGRAM, NotificationEvent.SUCCESS_OFFER)
+class TelegramSuccessOffer(BaseTemplator):
+    pass
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/notification/models.py:125]:
```suggestion
```

File: cryptostock_api/cryptostock/notification/models.py [hunk 3]
```diff
@@ -98,3 +99,27 @@ def enable_notification_subscriptions(self):
...
+
+    @classmethod
+    def register(cls, consumer_type, notification_type):
+        def inner(templater_cls):
+            cls._templaters[consumer_type, notification_type] = templater_cls
+            return templater_cls
+
+        return inner
+
+    get = classmethod(_templaters.get)
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/notification/models.py:115]: Should return `BaseTemplator`

File: cryptostock_api/cryptostock/notification/models.py [hunk 4]
```diff
@@ -98,3 +99,27 @@ def enable_notification_subscriptions(self):
...
+
+
+class TemplaterRegister:
+    _templaters = {}
+
+    @classmethod
+    def register(cls, consumer_type, notification_type):
+        def inner(templater_cls):
+            cls._templaters[consumer_type, notification_type] = templater_cls
+            return templater_cls
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/notification/models.py:111]:
```suggestion
            # TODO: register logic
            return templater_cls
```

**Review by vadim-denisov — CHANGES_REQUESTED** [review]

File: cryptostock_api/cryptostock/notification/models.py (hunk 3 above)
Comment (Lex-23) [reply cryptostock_api/cryptostock/notification/models.py:115]: got it

File: cryptostock_api/cryptostock/notification/models.py (hunk 2 above)
Comment (Lex-23) [reply cryptostock_api/cryptostock/notification/models.py:125]: ok

File: cryptostock_api/cryptostock/notification/models.py (hunk 1 above)
Comment (Lex-23) [reply cryptostock_api/cryptostock/notification/models.py:120]: ok

File: cryptostock_api/cryptostock/notification/models.py (hunk 3 above)
Comment (vadim-denisov) [reply cryptostock_api/cryptostock/notification/models.py:115]: Should ALWAYS return BaseRegister

File: cryptostock_api/cryptostock/notification/models.py [hunk 5]
```diff
@@ -98,3 +99,23 @@ def enable_notification_subscriptions(self):
...
+        def inner(templater_cls):
+            # TODO: register logic
+            return templater_cls
+
+        return inner
+
+    get = classmethod(_templaters.get)
+
+
+@TemplaterRegister.register(ConsumerType.TELEGRAM, NotificationEvent.SUCCESS_OFFER)
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/notification/models.py:118]:
```suggestion
```

**Review by vadim-denisov — CHANGES_REQUESTED** [review]

File: cryptostock_api/cryptostock/notification/models.py [hunk 6]
```diff
@@ -82,8 +83,8 @@ class Consumer(models.Model):
     def enable_consumers(self):
         return Consumer.objects.all().filter(account=self.account, enable=True)
 
-    def send(self, message):
-        return SENDER[self.type](message)
+    def send(self, message, **data):
+        return SENDER[self.type](message, **data)
```
Comment (vadim-denisov) [inline cryptostock_api/cryptostock/notification/models.py:87]:
```suggestion
    def send(self, **data):
        sender = SENDER.get(self.type)
        if sender is None:
            raise ValueError(f"Sender doesn't exist, type={self.type}")
        return sender(**data)
```
```

**Review by vadim-denisov — APPROVED** [review]

**Comment by Lex-23** [comment]
[Upgrading notification system](https://trello.com/c/rI1ECV4R/32-upgrading-notification-system)

**Pull request merged**